Use assert_implements in test the-input-element/type-change-state.html

clopez opened the pull request:
Use assert_precondition to check if the browser supports both input element types before trying to change the state from one to other.

For example, WebKitGTK and Safari doesn't support several of the input types tested here (like those related to dates). The output of the test its clearer if the precondition its checked before.

clopez commented:
Check how it looks on safari before and after this https://wpt.fyi/results/html/semantics/forms/the-input-element/type-change-state.html?diff&filter=ADC&run_id=444340020&run_id=444320020 (the precondition_failed makes it clear to see the results and the actual failures)

stephenmcgruer commented:
Are these areas of the spec `OPTIONAL`? See https://github.com/web-platform-tests/rfcs/pull/48 for an in-progress RFC where we intend to deprecate `assert_precondition` in favour of two methods; `assert_implements` (which maps to FAIL) and `assert_implements_optional` (which maps to PRECONDITION_FAILED for legacy reasons).

clopez commented:
> Are these areas of the spec `OPTIONAL`?

I don't think this are optional. 

> See [web-platform-tests/rfcs#48](https://github.com/web-platform-tests/rfcs/pull/48) for an in-progress RFC where we intend to deprecate `assert_precondition` in favour of two methods; `assert_implements` (which maps to FAIL) and `assert_implements_optional` (which maps to PRECONDITION_FAILED for legacy reasons).

Interesting, i'll wait for that to land and then update this PR to use ```assert_implements```

stephenmcgruer commented:
FYI `assert_implements` and `assert_implements_optional` are now available in WPT.

clopez committed:
Use assert_implements in test the-input-element/type-change-state.html

* Use assert_implements to check if the browser supports both input
element types before trying to change the state from one to other.

* For example, WebKitGTK and Safari doesn't support several of the input
types tested here (like those related to dates). The output of the test
its clearer if the precondition its checked before.

clopez commented:
Updated PR to use ```assert_implements```

annevk reviewed (approved):
This seems fine. Consider using `const` in future PRs.
